### PR TITLE
Added CircuitBrains Basic and Deluxe Boards

### PR DESCRIPTION
**Pull request by neubauek**
Requesting the addition of my 2 CircuitPython boards.

### CODE REVIEW

**Comment by ladyada** [comment]
yay! awesome :)

**Comment by ladyada** [comment]
ok ya just need to add the build lines to build.yml, goes in alphabetical

```Boards missing in .github/workflows/build.yml:
circuitbrains_deluxe_m4
circuitbrains_basic_m0
```

File: ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// These are pins not to reset.
+#define MICROPY_PORT_A        (0)
+#define MICROPY_PORT_B        (0)
+#define MICROPY_PORT_C        (0)
+
+// If you change this, then make sure to update the linker scripts as well to
+// make sure you don't overwrite code.
+#define CIRCUITPY_INTERNAL_NVM_SIZE 256
+
+#define BOARD_FLASH_SIZE (0x00040000 - 0x2000 - CIRCUITPY_INTERNAL_NVM_SIZE)
```
Comment (dhalbert) [inline ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.h]: You can delete this; it's not used in atmel-samd now.

File: ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#define SPI_FLASH_CS_PIN            &pin_PA19
+
+// These are pins not to reset.
+#define MICROPY_PORT_A        (0)
+#define MICROPY_PORT_B        (0)
+#define MICROPY_PORT_C        (0)
+
+// If you change this, then make sure to update the linker scripts as well to
+// make sure you don't overwrite code.
+#define CIRCUITPY_INTERNAL_NVM_SIZE 256
```
Comment (dhalbert) [inline ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.h]: This is now the default. You can remove these lines.

File: ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.mk [hunk 3]
```diff
@@ -0,0 +1,17 @@
+LD_FILE = boards/samd21x18-bootloader-external-flash-crystalless.ld
```
Comment (dhalbert) [inline ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.mk]: Obsolete; remove `LD_FILE` definition.

File: ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.mk [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+CHIP_FAMILY = samd21
+
+SPI_FLASH_FILESYSTEM = 1
+EXTERNAL_FLASH_DEVICE_COUNT = 1
+EXTERNAL_FLASH_DEVICES = "W25Q32JV_IQ"
+LONGINT_IMPL = MPZ
+
+CIRCUITPY_BITBANGIO = 0
+CIRCUITPY_FREQUENCYIO = 0
+CIRCUITPY_I2CSLAVE = 0
```
Comment (dhalbert) [inline ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.mk]: Add newline to end of file.

File: ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.h [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+// QSPI CS, QSPI SCK
+#define MICROPY_PORT_B (PORT_PB10 | PORT_PB11)
+#define MICROPY_PORT_C (0)
+#define MICROPY_PORT_D (0)
+
+#define AUTORESET_DELAY_MS 500
+
+// If you change this, then make sure to update the linker scripts as well to
+// make sure you don't overwrite code
+#define CIRCUITPY_INTERNAL_NVM_SIZE 8192
```
Comment (dhalbert) [inline ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.h]: This is the default on M4; you can remove these lines.

File: ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.h [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+#define BOARD_FLASH_SIZE (FLASH_SIZE - 0x4000 - CIRCUITPY_INTERNAL_NVM_SIZE)
```
Comment (dhalbert) [inline ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.h]: No longer used for atmel-samd; remove.

File: ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.h [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+#define DEFAULT_SPI_BUS_SCK (&pin_PA13)
+#define DEFAULT_SPI_BUS_MOSI (&pin_PA12)
+#define DEFAULT_SPI_BUS_MISO (&pin_PA14)
+
+#define DEFAULT_UART_BUS_RX (&pin_PA23)
+#define DEFAULT_UART_BUS_TX (&pin_PA22)
+
+// USB is always used internally so skip the pin objects for it.
+#define IGNORE_PIN_PA24     1
+#define IGNORE_PIN_PA25     1
```
Comment (dhalbert) [inline ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.h]: Add newline to end of file.

File: ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.mk [hunk 8]
```diff
@@ -0,0 +1,17 @@
+LD_FILE = boards/samd51x19-bootloader-external-flash.ld
```
Comment (dhalbert) [inline ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.mk]: Obsolete; remove.

File: ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.mk [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+CHIP_FAMILY = samd51
+
+QSPI_FLASH_FILESYSTEM = 1
+EXTERNAL_FLASH_DEVICE_COUNT = 2
+EXTERNAL_FLASH_DEVICES = "W25Q64JV_IQ, S25FL064L"
+LONGINT_IMPL = MPZ
+
+CIRCUITPY_NETWORK = 1
+MICROPY_PY_WIZNET5K = 5500
+CIRCUITPY_PS2IO = 1
```
Comment (dhalbert) [inline ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.mk]: Add newline to end of file.

File: ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.h (hunk 1 above)
Comment (neubauek) [reply ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.h]: Cleaned up board definitions per this conversation.

File: ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.mk (hunk 4 above)
Comment (neubauek) [reply ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.mk]: Cleaned up board definitions per this conversation.

File: ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.h (hunk 6 above)
Comment (neubauek) [reply ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.h]: Cleaned up board definitions per this conversation.

File: ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.mk (hunk 3 above)
Comment (neubauek) [reply ports/atmel-samd/boards/circuitbrains_basic_m0/mpconfigboard.mk]: Cleaned up board definitions per this conversation.

File: ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.h (hunk 5 above)
Comment (neubauek) [reply ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.h]: Cleaned up board definitions per this conversation.

File: ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.h (hunk 7 above)
Comment (neubauek) [reply ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.h]: Cleaned up board definitions per this conversation.

File: ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.mk (hunk 8 above)
Comment (neubauek) [reply ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.mk]: Cleaned up board definitions per this conversation.

File: ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.mk (hunk 9 above)
Comment (neubauek) [reply ports/atmel-samd/boards/circuitbrains_deluxe_m4/mpconfigboard.mk]: Cleaned up board definitions per this conversation.

**Pull request merged**